pkg/datasources/external_functions.go

```diff
@@ -1,24 +1,25 @@
 package datasources
 
 import (
+	"context"
 	"database/sql"
-	"errors"
 	"fmt"
-	"log"
+	"strings"
 
-	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/snowflake"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
 var externalFunctionsSchema = map[string]*schema.Schema{
 	"database": {
 		Type:        schema.TypeString,
-		Required:    true,
+		Optional:    true,
 		Description: "The database from which to return the schemas from.",
 	},
 	"schema": {
 		Type:        schema.TypeString,
-		Required:    true,
+		Optional:    true,
```

Comment:
database should be required if schema is present. maybe add requiredWith?

Reply:
done, and added to MIGRATION_GUIDE